src/arviz_plots/plots/essplot.py

```diff
@@ -0,0 +1,251 @@
+        sample_dims = rcParams["data.sample_dims"]
+    if isinstance(sample_dims, str):
+        sample_dims = [sample_dims]
+
+    # mutable inputs
+    if plot_kwargs is None:
+        plot_kwargs = {}
+    if pc_kwargs is None:
+        pc_kwargs = {}
+    else:
+        pc_kwargs = pc_kwargs.copy()
+
+    if stats_kwargs is None:
+        stats_kwargs = {}
+
+    # processing dt/group/coords/filtering
+    distribution = process_group_variables_coords(
+        dt, group=group, var_names=var_names, filter_vars=filter_vars, coords=coords
+    )
+
+    # set plot collection initialization defaults if it doesnt exist
+    if plot_collection is None:
+        if backend is None:
+            backend = rcParams["plot.backend"]
+        pc_kwargs.setdefault("col_wrap", 5)
+        pc_kwargs.setdefault(
+            "cols",
+            ["__variable__"]
+            + [dim for dim in distribution.dims if dim not in {"model"}.union(sample_dims)],
+        )
+        if "model" in distribution:
+            pc_kwargs["aes"] = pc_kwargs.get("aes", {}).copy()
+            pc_kwargs["aes"].setdefault("color", ["model"])
+        plot_collection = PlotCollection.wrap(
+            distribution,
+            backend=backend,
+            **pc_kwargs,
+        )
+
+    # set plot collection dependent defaults (like aesthetics mappings for each artist)
+    if aes_map is None:
+        aes_map = {}
+    else:
+        aes_map = aes_map.copy()
+    aes_map.setdefault(kind, plot_collection.aes_set)
+    if labeller is None:
+        labeller = BaseLabeller()
+
+    # compute and add ess subplots
+    # step 1
+    ess_kwargs = copy(plot_kwargs.get(kind, {}))
+
+    if ess_kwargs is not False:
+        # step 2
+        ess_dims, _, ess_ignore = filter_aes(plot_collection, aes_map, kind, sample_dims)
+        if kind == "local":
+            probs = np.linspace(0, 1, n_points, endpoint=False)
+            xdata = probs
+
+            # step 3
+            ess_y_dataset = xr.concat(
+                [
+                    distribution.azstats.ess(
+                        dims=ess_dims,
+                        method="local",
+                        relative=relative,
+                        prob=[p, (p + 1 / n_points)],
+                        **stats_kwargs.get("ess", {}),
+                    )
+                    for p in probs
+                ],
+                dim="ess_dim",
+            )
+            print(f"\n ess_y_dataset = {ess_y_dataset}")
+
+            # broadcasting xdata to match ess_y_dataset's shape
+            xdata_da = xr.DataArray(xdata, dims="ess_dim")
+            print(f"\n xdata_da ={xdata_da}")
+
+            # broadcasting xdata_da to match shape of each variable in ess_y_dataset
+            xdata_broadcasted_dict = {}
+            for var in ess_y_dataset.data_vars:
+                _, xdata_broadcasted = xr.broadcast(ess_y_dataset[var], xdata_da)
+                xdata_broadcasted_dict[var] = xdata_broadcasted
```

Comment:
```suggestion
            xdata_broadcasted_dict = {var_name: xdata_da.broadcast_like(da) for var_name, da in ess_y_dataset.items()}
```

This should generate the same dictionary being slightly more efficient

could also be merged with the `xdata_dataset` line so it is `xdata_dataset = xr.Dataset({var_name: ...})`